src/components/candevice/candevice.cpp

```diff
@@ -55,6 +78,11 @@ void CanDevice::sendFrame(const QCanBusFrame& frame)
     }
 }
 
+ComponentInterface::ComponentProperties CanDevice::getSupportedProperties() const
+{
+    return d_ptr->_supportedProps;
```

Comment:
Does `Q_D` checks `d_ptr` against `nullptr`?

```
Q_D(CanDevice);

return d->...
```

Reply:
I don't think so. But it's used a lot in our code and never checked.

Reply:
So we shall start checking that immidiately :-)

Reply:
Could you provide a snippet of code that creates any of these classes with d_ptr == nullptr? As far as I can tell from the code that is impossible, so there is no point in checking.